editoast/src/converters/utils.rs

```diff
@@ -0,0 +1,213 @@
+                "SOUTH-2".into(),
+                "NORTH-1".into(),
+                "NORTH-2".into(),
+            ],
+            groups: double_cross_groups,
+        },
+    ]
+}
+
+pub enum EdgeAngleError {
+    InvalidAngle(NodeId),
+}
+
+fn build_track_endpoint((edge, endpoint): (&Edge, Endpoint)) -> TrackEndpoint {
+    TrackEndpoint {
+        track: edge.id.as_str().into(),
+        endpoint,
+    }
+}
+
+pub fn build_track_section_link(
+    edge_a: (&Edge, Endpoint),
+    edge_b: (&Edge, Endpoint),
+) -> TrackSectionLink {
+    let id = if edge_a.1 == Endpoint::Begin {
+        edge_a.0.source
+    } else {
+        edge_a.0.target
+    };
+    let src = build_track_endpoint(edge_a);
+    let dst = build_track_endpoint(edge_b);
+    TrackSectionLink {
+        id: id.0.to_string().into(),
+        src,
+        dst,
+    }
+}
+
+// Given an edge and it’s enpoint, returns the coordinates used to compute the angle
+// It uses the nearest OpenStreetMap node, and the other as the the rails might do a loop
+// that would result in a bad angle
+fn reference_coord((edge, endpoint): (&Edge, Endpoint)) -> Coord {
+    if endpoint == Endpoint::Begin {
+        edge.geometry[1]
+    } else {
+        edge.geometry[edge.geometry.len() - 2]
+    }
+}
+
+// In order for a train to be able to go from one edge to another
+// The angle must be as flat as possible (180°)
+fn flat(angle: f64) -> bool {
+    (180.0 - angle).abs() <= 30.0
+}
+
+pub fn build_point_switch(
+    (node, edges): (&NodeId, &Vec<(&Edge, Endpoint)>),
+) -> Result<Switch, EdgeAngleError> {
+    let center = if edges[0].1 == Endpoint::Begin {
+        edges[0].0.geometry[0]
+    } else {
+        edges[0].0.geometry[edges[0].0.geometry.len() - 1]
+    };
+    let a = reference_coord(edges[0]);
+    let b = reference_coord(edges[1]);
+    let c = reference_coord(edges[2]);
+
+    let ab = angle(center, a, b);
+    let ac = angle(center, a, c);
+    let bc = angle(center, b, c);
+
+    let track_endpoint_a = build_track_endpoint(edges[0]);
+    let track_endpoint_b = build_track_endpoint(edges[1]);
+    let track_endpoint_c = build_track_endpoint(edges[1]);
+
+    let (base, left, right) = match (flat(ab), flat(ac), flat(bc)) {
+        (true, true, false) => Ok((track_endpoint_a, track_endpoint_b, track_endpoint_c)),
+        (true, false, true) => Ok((track_endpoint_b, track_endpoint_a, track_endpoint_c)),
+        (false, true, true) => Ok((track_endpoint_c, track_endpoint_a, track_endpoint_b)),
+        _ => {
+            eprintln!("point switch {} impossible angles {ab}, {ac}, {bc}", node.0);
+            Err(EdgeAngleError::InvalidAngle(*node))
+        }
+    }?;
+
+    let mut ports = HashMap::new();
+    ports.insert("BASE".into(), base);
+    ports.insert("LEFT".into(), left);
+    ports.insert("RIGHT".into(), right);
+
+    Ok(Switch {
+        id: node.0.to_string().into(),
+        switch_type: "point".into(),
+        ports,
+        ..Default::default()
+    })
```

Comment:
We can set the `group_change_delay` to `4.`. This is a more realistic approximation.